net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -39,85 +39,50 @@ NO_IPV6=$?
 
 # return true(=0) if has any mwan3 interface enabled
 # otherwise return false
-mwan3_rtmon_ipv4()
+mwan3_rtmon_route()
 {
-	local idx=0
+	config_load mwan3
 	local ret=1
-	local tbl=""
-
-	local tid family enabled
-
+	local tid=0
+	local family enabled IP line
 	mkdir -p /tmp/mwan3rtmon
 	($IP4 route list table main  | grep -v "^default\|linkdown" | sort -n; echo empty fixup) >/tmp/mwan3rtmon/ipv4.main
-	while uci get mwan3.@interface[$idx] >/dev/null 2>&1 ; do
-		tid=$((idx+1))
-
-		family="$(uci -q get mwan3.@interface[$idx].family)"
-		[ -z "$family" ] && family="ipv4"
-
-		enabled="$(uci -q get mwan3.@interface[$idx].enabled)"
-		[ -z "$enabled" ] && enabled="0"
+	[ $NO_IPV6 -eq 0 ] && ($IP6 route list table main  | grep -v "^default\|^::/0\|^fe80::/64\|^unreachable" | sort -n; echo empty fixup) >/tmp/mwan3rtmon/ipv6.main
+	config load mwan3
+	for section in ${CONFIG_SECTIONS}; do
```

Comment:
We could move this for section into a function and could then do a config_foreach call.
See https://github.com/openwrt/packages/blob/master/net/mwan3/files/lib/mwan3/mwan3.sh#L217

Reply:
fixed in 169982d. We can squash everything into one commit before merging.

Reply:
Fixed in 3543df28e.